agent/ui_endpoint.go

```diff
@@ -155,7 +160,8 @@ RPC:
 	}
 
 	// Generate the summary
-	return summarizeServices(out.Nodes), nil
+	// TODO (gateways) (freddy) Have Internal.ServiceDump return ServiceDump instead. Need to add bexpr filtering for type.
```

Comment:
Actually, not sure anymore if that's the right direction. It would make the ACL filtering weirder. Since some cases won't have service/node info and other won't have gateway info.

Reply:
Yeah I see what you are saying. I think there is something to converging on a shared data structure (esp. one named ServiceDump), but probably some finicky details to consider there.